app/src/UI/Features/LegacyMap/helpers/components/DrawControls.tsx

```diff
@@ -38,7 +45,8 @@ const DrawControls = () => {
   const tileCacheMode = useSelector((state) => state.Map.tileCacheMode);
   const drawingCustomLayer = useSelector((state) => state.Map.drawingCustomLayer);
   const appModeURL = useSelector((state) => state.AppMode.url);
-
+  const currGeoTrackingMode = useSelector((state) => state.Map.track_me_draw_geo.drawingShape);
+  const [prevGeoTrackingMode, setPrevGeoTrackingMode] = useState<boolean | undefined>(undefined);
```

Comment:
```suggestion
  const [prevGeoTrackingMode, setPrevGeoTrackingMode] = useState<boolean>(false);
```
boolean is never set back to undefined or checked for undefined status. undefined == false